### PR TITLE
ValidFilter + value in errors

### PR DESCRIPTION
**Pull request by rpiaggio**
- Introduce a `ValidFilter` optic, composable with `ValidWedge` and `ValidSplitEpi`, to perform simple validations without transforming the value. Allows composing a `ValidSplitEpi` parser validator with, for example, checking that a parsed numeric value is within certain bounds.
- Add the input value as a parameter to the error instances. Allows error messages like `Value XXX is out of bounds`.

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/math/units.scala [hunk 1]
```diff
@@ -142,6 +142,8 @@ trait units {
 
   given ValueConversion[Parallax.LongParallaxμas, Rational] = Rational(_)
 
+  given refinedValueConversion[V, P]: ValueConversion[V Refined P, V] = _.value
```
Comment (rpiaggio) [inline modules/core/shared/src/main/scala/lucuma/core/math/units.scala:145]: This is unrelated. Allows easy "unrefining" of coulomb quantities.

File: modules/core/shared/src/main/scala/lucuma/core/optics/ValidFormat.scala [hunk 2]
```diff
@@ -13,4 +13,10 @@ trait ValidFormat[E, A, B] {
   val getValid: A => Either[E, B]
 
   val reverseGet: B => A
+
+  /** Like getValid, but throws IllegalArgumentException when Invalid. */
+  def unsafeGet(a: A): B =
```
Comment (rpiaggio) [inline modules/core/shared/src/main/scala/lucuma/core/optics/ValidFormat.scala:18]: Pulled upwards from subtypes.

**Review by toddburnside — APPROVED** [review]
LGTM
